add gtk4/adw dependencies

julianmueller opened the pull request:
<!-- Thank you for contributing a change to the rosdistro. There are two primary types of submissions.
Please select the appropriate template from below: ROSDEP_RULE_TEMPLATE or DOC_INDEX_TEMPLATE

If you're making a new release with bloom please use bloom to create the pull request automatically (except for the naming review request which must be made manually).
If you've already run the release bloom has a `--pull-request-only` option you can use.-->

<!-- ROSDEP_RULE_TEMPLATE: Submitter Please review the contributing guidelines: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md -->

Please add the following dependency to the rosdep database.

## Package name:

1. gtk4
2. libgirepository-dev
3. adwaita-icon-theme
4. libadwaita-dev

## Package Upstream Source:

1. https://gitlab.gnome.org/GNOME/gtk
2. https://gitlab.gnome.org/GNOME/gobject-introspection
3. https://gitlab.gnome.org/GNOME/adwaita-icon-theme
4. https://gitlab.gnome.org/GNOME/libadwaita

## Purpose of using this:

- used in the gtk4/adw gui: https://github.com/julianmueller/insight_gui/

Distro packaging links:

## Links to Distribution Packages

<!-- Replace the REQUIRED areas with the URL to the package.  For IF AVAILABLE areas, either put in the URL to the package or state 'not available'.
More info at https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#guidelines-for-rosdep-rules -->

- Debian: https://packages.debian.org/
  - 1. libgtk4-dev
  - 2. libgirepository1.0-dev
  - 3. adwaita-icon-theme
  - 4. libadwaita-1-dev
- Ubuntu: https://packages.ubuntu.com/
  - 1. libgtk4-dev
  - 2. libgirepository1.0-dev
  - 3. adwaita-icon-theme
  - 4. libadwaita-1-dev

## Additional

I also added `lark` to the requirements.txt, because my pytest failed as this was missing.

Commit:
add gtk4/adw dependencies

julianmueller commented:
@clalancette @MichaelOrlov Sorry to bother you two, but you just recently approved a PR on here. Would you be so kind to review my PR? I have a pkg at rosdistro, that is in need of these dependencies. Thanks!

sloretz committed:
Remove lark

Removing Lark because it's unrelated to the added rosdep keys

sloretz commented on rosdep/base.yaml:
```diff
@@ -1893,6 +1901,17 @@ gtk3:
     slackpkg:
       packages: [gtk+3]
   ubuntu: [libgtk-3-dev]
+gtk4:
```
Resolving since this matches gtk2 and gtk3 rosdep keys already

sloretz commented on rosdep/base.yaml:
```diff
@@ -2631,6 +2650,15 @@ libadolc-dev:
   nixos: [adolc]
   opensuse: [adolc-devel]
   ubuntu: [libadolc-dev]
+libadwaita-dev:
```
> If a native package is available, the key name should match the name of the package in Ubuntu.

Technically the rule says keys must match the Debian/Ubuntu package name, but None of the other distros are including the version in the package name so I'm going to resolve this one.

sloretz commented on rosdep/base.yaml:
```diff
@@ -4026,6 +4054,13 @@ libgif-dev:
   gentoo: [media-libs/giflib]
   nixos: [giflib]
   ubuntu: [libgif-dev]
+libgirepository-dev:
```
> If a native package is available, the key name should match the name of the package in Ubuntu.

Technically the rule says keys must match the Debian/Ubuntu package name, but none of the other distros are including the version in the package name so I'm going to resolve this one.

sloretz commented on rosdep/base.yaml:
```diff
@@ -2631,6 +2650,15 @@ libadolc-dev:
   nixos: [adolc]
   opensuse: [adolc-devel]
   ubuntu: [libadolc-dev]
+libadwaita-dev:
```
> If a native package is available, the key name should match the name of the package in Ubuntu.

Technically the rule says keys must match the Debian/Ubuntu package name, but none of the other distros are including the version in the package name so I'm going to resolve this one.

sloretz commented on rosdep/base.yaml:
```diff
@@ -4040,6 +4068,13 @@ libgif-dev:
   gentoo: [media-libs/giflib]
   nixos: [giflib]
   ubuntu: [libgif-dev]
+libgirepository-dev:
```
> If a native package is available, the key name should match the name of the package in Ubuntu.

Technically the rule says keys must match the Debian/Ubuntu package name, but none of the other distros are including the version in the package name so I'm going to resolve this one.

julianmueller committed:
Update rosdep/base.yaml

Co-authored-by: Shane Loretz <shane.loretz@gmail.com>

julianmueller committed:
Update rosdep/base.yaml

Co-authored-by: Shane Loretz <shane.loretz@gmail.com>

julianmueller committed:
Update rosdep/base.yaml

Co-authored-by: Shane Loretz <shane.loretz@gmail.com>

julianmueller commented on rosdep/base.yaml:
```diff
@@ -1893,6 +1901,17 @@ gtk3:
+  openembedded: [gtk4@openembedded-core]
+  ubuntu:
+    '*': [libgtk-4-dev]
+    focal: null
```
I guess rhel 8 must be excluded, like I did with ubuntu

julianmueller commented on rosdep/base.yaml:
```diff
@@ -2631,6 +2650,15 @@ libadolc-dev:
+  openembedded: [libadwaita@openembedded-core]
+  ubuntu:
+    '*': [libadwaita-1-dev]
+    focal: null
```
I guess rhel 8 must be excluded, like I did with ubuntu

julianmueller committed:
Update rosdep/base.yaml

Co-authored-by: Steven! Ragnarök <nuclearsandwich@users.noreply.github.com>

julianmueller committed:
Update rosdep/base.yaml

Co-authored-by: Steven! Ragnarök <nuclearsandwich@users.noreply.github.com>

julianmueller committed:
Update base.yaml

ERR: list out of alphabetical order line 1917.  '*' should come before '8'
ERR: list out of alphabetical order line 2668.  '*' should come before '8'

julianmueller committed:
Update base.yaml

remove opensuse for adwaita-icon-theme

julianmueller committed:
Update base.yaml

remove added distros from @sloretz

julianmueller committed:
Update base.yaml

julianmueller commented:
I dont know why it keeps failing. Its not like I havent tried... Maybe @nuclearsandwich can enlighten me?

cottsay commented:
There is something wrong with the decompression mechanism in rosdep_repo_check. I can reproduce it locally with `http://azure.repo.almalinux.org/9.6/CRB/x86_64/os/` as a baseurl.

The GitHub Actions runner is choosing a different repository mirror than your local runs, which is why it's tough to reproduce locally. I'm not sure I can get this fixed this afternoon, but I should have cycles to have a look tonight.

julianmueller commented:
> There is something wrong with the decompression mechanism in rosdep_repo_check. I can reproduce it locally with `http://azure.repo.almalinux.org/9.6/CRB/x86_64/os/` as a baseurl.
> 
> The GitHub Actions runner is choosing a different repository mirror than your local runs, which is why it's tough to reproduce locally. I'm not sure I can get this fixed this afternoon, but I should have cycles to have a look tonight.

Thanks! Do you have an update on this?

julianmueller committed:
Merge branch 'master' into master

quarkytale committed:
Merge branch 'master' into master

julianmueller commented:
@quarkytale and @emersonknapp thanks for the approval! But how to proceed now? Can the PR be merged? Why is there still the "rosdep-needs-second-approval" tag?

Commit:
Merge branch 'master' into master

julianmueller committed:
Merge branch 'master' into master

nuclearsandwich committed:
Merge branch 'master' into master
